### PR TITLE
Feat/add data on execute callback

### PR DESCRIPTION
**Pull request by jmanrumartinez**
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Only
- [ ] Refactor

### Description, Motivation and Context

We need to know which action occurred on each call __callbackPhotosUploaded_ so we thought that could be fine to pass some extra data whit it.

### CODE REVIEW

File: components/molecule/photoUploader/src/index.js [hunk 1]
```diff
@@ -116,7 +117,7 @@ const MoleculePhotoUploader = forwardRef(
       }
     })
 
-    const _callbackPhotosUploaded = list => {
+    const _callbackPhotosUploaded = (list, data) => {
```
Comment (jmanrumartinez) [inline components/molecule/photoUploader/src/index.js]: Not sure about the `data` naming, it was harder for me to find the proper name for it, `data` is an object with info why occurred the callback _(maybe fallback (?) but it's a contradiction)_, if you have any suggestions, tell me! Thank you!

File: components/molecule/photoUploader/src/PhotosPreview.js [hunk 2]
```diff
@@ -48,16 +49,22 @@ const PhotosPreview = ({
...
     const list = [...files]
     list.splice(index, 1)
     setFiles(list)
     setNotificationError(DEFAULT_NOTIFICATION_ERROR)
-    _callbackPhotosUploaded(list)
+    _callbackPhotosUploaded(list, {
+      action: ACTIONS.DELETE,
+      data: {
+        itemIndex: index,
+        item: files[index]
```
Comment (arnau-rius) [inline components/molecule/photoUploader/src/PhotosPreview.js:65]: `itemDeleted`?

File: components/molecule/photoUploader/src/PhotosPreview.js [hunk 3]
```diff
@@ -48,16 +49,22 @@ const PhotosPreview = ({
...
   const _deleteItem = index => {
     const list = [...files]
     list.splice(index, 1)
     setFiles(list)
     setNotificationError(DEFAULT_NOTIFICATION_ERROR)
-    _callbackPhotosUploaded(list)
+    _callbackPhotosUploaded(list, {
+      action: ACTIONS.DELETE,
+      data: {
+        itemIndex: index,
```
Comment (arnau-rius) [inline components/molecule/photoUploader/src/PhotosPreview.js:64]: `itemDeletedIndex`?

File: components/molecule/photoUploader/src/PhotosPreview.js [hunk 4]
```diff
@@ -179,7 +198,7 @@ const PhotosPreview = ({
       setList={setFiles}
       animation={200}
       draggable={`.${THUMB_SORTABLE_CLASS_NAME}`}
-      onEnd={() => _onSortEnd()}
+      onEnd={sortedFile => _onSortEnd(sortedFile)}
```
Comment (arnau-rius) [inline components/molecule/photoUploader/src/PhotosPreview.js]: Just for consistency with _onSortEnd func argument naming
```suggestion
      onEnd={event => _onSortEnd(event)}
```

File: components/molecule/photoUploader/src/config.js [hunk 5]
```diff
@@ -40,3 +40,12 @@ export const REJECT_FILES_REASONS = {
...
   loadFailed: 'Load failed, corrupt file.'
 }
+
+export const ACTIONS = {
+  SORT: 'SORT',
+  DELETE: 'DELETE',
+  ROTATE: 'ROTATE',
+  RETRY_UPLOAD: 'RETRY_UPLOAD',
+  UPLOAD: 'UPLOAD',
+  LOAD_INITIAL_PHOTOS: 'LOAD_INITIAL_PHOTOS'
```
Comment (arnau-rius) [inline components/molecule/photoUploader/src/config.js]: `INITIAL_LOAD`?

File: components/molecule/photoUploader/src/index.js (hunk 1 above)
Comment (arnau-rius) [reply components/molecule/photoUploader/src/index.js]: Since it is generic depending the action, `data` looks good to me.

**Review by arnau-rius — APPROVED** [review]

**Review by adriadoz — APPROVED** [review]

**Review by javiauso — APPROVED** [review]

**Review by andresin87 — APPROVED** [review]

**Pull request merged**